app/src/main/java/com/github/multimatum_team/multimatum/activity/ProfileActivity.kt

```diff
@@ -34,8 +38,12 @@ class ProfileActivity : AppCompatActivity() {
 
         logOutButton.setOnClickListener {
             Log.d(TAG, "Logging out...")
-            userViewModel.signOut()
-            AuthUI.getInstance().signOut(this)
+            if((this.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager).activeNetwork != null) {
+                userViewModel.signOut()
+                AuthUI.getInstance().signOut(this)
+            }else{
+                Toast.makeText(this, "You cannot log out without internet connection", Toast.LENGTH_SHORT).show()
```

Comment:
Maybe make an alertDialog instead of a toast, it'ld make more sense I think

Reply:
I think not because with alert dialog you have 2 button that doesn't have any purpose in this case so a toast seems to be a good solution in this case.